lib/classes/CLI.js

```diff
@@ -252,11 +252,14 @@ class CLI {
 
     // Throw error if command not found.
     if (!command) {
-      const errorMessage = [
-        `Serverless command "${commandName}" not found.`,
-        ' Run "serverless help" for a list of all available commands.',
-      ].join('');
-      throw new this.serverless.classes.Error(errorMessage);
+      return getCommandSuggestion(commandName, this.serverless.cli.loadedCommands)
+        .then(suggestedCommand => {
+          const errorMessage = [
+            `Serverless command "${commandName}" not found. Did you mean "${suggestedCommand}"?`,
+            ' Run "serverless help" for a list of all available commands.',
+          ].join('');
+          throw new this.serverless.classes.Error(errorMessage);
```

Comment:
We should not throw within promise chains, but consistently reject.
```
return BbPromise.reject(new this.serverless.classes.Error(errorMessage));
```
Having synchronous and asyncronous handling mixed makes it very hard to have consistent behavior and unit tests. A method that returns a promise on success should consistently reject on failure.
I know there are many occasions where it is not done, but we should adapt any occurence in modified code.